### PR TITLE
improve: various add-ons: Even More SAST (sonar) fixes

### PR DESCRIPTION
**Pull request by kingthorin**
- ascanrulesBeta > HiddenFlieScanRuleUnitTest, ensure unit tests have an assertion.
- "JUnit5 test classes and methods should have default package visibility"
  - Automation
  - commonlib
  - Kotlin
  - pscanrules
  - soap

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/ascanrules/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 - Correct Context check in SQL Injection scan rule.
+- Java compatibility issues with Source Code Disclosure WEB-INF scan rule and newer JREs (Issue 4038).
```
Comment (thc202) [inline addOns/ascanrules/CHANGELOG.md:12]: It needs to undo zaproxy/zap-extensions#1723. Probably better to do in its own PR or commit.

Comment (kingthorin) [reply addOns/ascanrules/CHANGELOG.md:12]: Okay

Comment (kingthorin) [reply addOns/ascanrules/CHANGELOG.md]: Split into: https://github.com/zaproxy/zap-extensions/pull/2959

**Comment by thc202** [comment]
The pull request description needs to be updated.

**Comment by kingthorin** [comment]
Thanks, done.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by thc202** [comment]
btw, were these introduced in the meantime or missed when doing the whole change?

**Review by psiinon — APPROVED** [review]

**Comment by kingthorin** [comment]
I’m not entirely sure. I believe the SOAP stuff was being changed at about the same time. Common lib was previous skipped due to dependency issues, which were addressed in the meantime. I saw one occurrence for sure that would not have matched the regex searches I’d done.

**Comment by thc202** [comment]
OK, thank you.
